### PR TITLE
CNF-14784: CNF-14787: Examples & Steps for Day 2 ACM PG change

### PR DESCRIPTION
**Pull request by irinamihai**
Description:
- Provide readme and examples to showcase the needed steps for the Day 2 operation when an existing ACM PG is updated:
  * ACM PG v2
  * ClusterInstance default ConfigMap v2
  * ClusterTemplate v2

### CODE REVIEW

File: docs/cluster_configuration.md [hunk 1]
```diff
@@ -177,3 +177,83 @@ status:
...
+        * The desired manifest section is updated. In the current samples [sno-ran-du-pg-v4-16-v2](#./git-setup/policytemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v2.yaml) adds a `sysctl` section to the `TunedPerformancePatch` section under the `v2-tuned-configuration-policy` policy.
+    * Create a new version of the [clusterinstance-defaults-v1](#./git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/clusterinstance-defaults-v1.yaml): [clusterinstance-defaults-v2](#./git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/clusterinstance-defaults-v2.yaml):
+        * Update the name to `clusterinstance-defaults-v2` (the namespace stays `sno-ran-du-v4-16`).
+        * Update the `sno-ran-du-policy` ManagedCluster extraLabel from `v1` to `v2`.
+    * Create a new version of the [sno-ran-du.v4-Y-Z-1](#./git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v1.yaml) `ClusterTemplate` CR: [sno-ran-du.v4-Y-Z-2](#./git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v2.yaml)
+        * Update the `metadata.name` from `sno-ran-du.v4-Y-Z-1` to `sno-ran-du.v4-Y-Z-1`
```
Comment (edcdavid) [inline docs/cluster_configuration.md:194]: typo, change to Update the `metadata.name` from `sno-ran-du.v4-Y-Z-1` to `sno-ran-du.v4-Y-Z-2`

File: docs/cluster_configuration.md [hunk 2]
```diff
@@ -177,3 +177,83 @@ status:
...
+        * Update `spec.version` from `v4-Y-Z-1` to `v4-Y-Z-1`
```
Comment (edcdavid) [inline docs/cluster_configuration.md:195]: same here

File: docs/samples/git-setup/policytemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v2.yaml [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                [bootloader]
+                cmdline_crash=-tsc=nowatchdog
+                cmdline_crash1=tsc=reliable
+                [sysctl]
+                kernel.timer_migration=1
+                kernel.sysrq=1
+                kernel.panic_on_rcu_stall=1
+                kernel.hung_task_panic=1
+                [scheduler]
+                [scheduler]
```
Comment (bartwensley) [inline docs/samples/git-setup/policytemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v2.yaml:129]: duplicate

File: docs/cluster_configuration.md [hunk 4]
```diff
@@ -177,3 +177,83 @@ status:
...
+1. Upversion the cluster template content:
+    * Create a new version of the ACM PG: [sno-ran-du-pg-v4-16-v2](#./git-setup/policytemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v2.yaml):
+        * The name is updated to `sno-ran-du-pg-v4-16-v2` (the `ztp-sno-ran-du-v4-16` namespace is kept).
+        * `policyDefaults.labelSelector.sno-ran-du-policy` is updated from `v1` to `v2` such that the policy binding is updates.
+        * All policy names are updated from `v1` to `v2` (example: `v1-subscriptions-policy` -> `v2-subscriptions-policy`).
+        * The desired manifest section is updated. In the current samples [sno-ran-du-pg-v4-16-v2](#./git-setup/policytemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v2.yaml) adds a `sysctl` section to the `TunedPerformancePatch` section under the `v2-tuned-configuration-policy` policy.
+    * Create a new version of the [clusterinstance-defaults-v1](#./git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/clusterinstance-defaults-v1.yaml): [clusterinstance-defaults-v2](#./git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/clusterinstance-defaults-v2.yaml):
```
Comment (bartwensley) [inline docs/cluster_configuration.md:190]: Links not working - do they need "samples" in the path?

File: docs/cluster_configuration.md [hunk 5]
```diff
@@ -177,3 +177,83 @@ status:
...
+        * Update the name to `clusterinstance-defaults-v2` (the namespace stays `sno-ran-du-v4-16`).
+        * Update the `sno-ran-du-policy` ManagedCluster extraLabel from `v1` to `v2`.
+    * Create a new version of the [sno-ran-du.v4-Y-Z-1](#./git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v1.yaml) `ClusterTemplate` CR: [sno-ran-du.v4-Y-Z-2](#./git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v2.yaml)
```
Comment (bartwensley) [inline docs/cluster_configuration.md:193]: It looks like the sno-ran-du.v4-Y-Z-1 isn't in git and the sno-ran-du-v2.yaml file needs to be renamed to match the new naming.

File: docs/cluster_configuration.md [hunk 6]
```diff
@@ -177,3 +177,83 @@ status:
...
+        * Update the `metadata.name` from `sno-ran-du.v4-Y-Z-1` to `sno-ran-du.v4-Y-Z-2`
+        * Update `spec.version` from `v4-Y-Z-1` to `v4-Y-Z-2`
+        * Update `spec.templates.clusterInstanceDefaults` to `clusterinstance-defaults-v2`
+2. ArgoCD sync to the hub cluster:
+    * Add the newly created files to their corresponding kustomization.yaml.
+    * All the resources created from above are created on the hub cluster, including the `v2` policies and the new ClusterTemplate is validated.
+    * The new policies are not yet applied to the cluster because the `ManagedCluster` still has the old `sno-ran-du-policy: "v1"` label
+3. The SMO selects the new ClusterTemplate CR for the ProvisioningRequest:
+    * `spec.templateVersion` is updated from `sno-ran-du.v4-Y-Z-1` to `sno-ran-du.v4-Y-Z-2`
```
Comment (bartwensley) [inline docs/cluster_configuration.md:202]: Would it be:
spec.templateName = sno-ran-du
spec.templateVersion - 4-Y-Z-1 -> 4-Y-Z-2

File: docs/cluster_configuration.md [hunk 7]
```diff
@@ -177,3 +177,83 @@ status:
...
+4. The IMS operator detects the change
+    * It updates the ClusterInstance with the updated `sno-ran-du-policy: "v2"` ManagedCluster label
+    * the siteconfig operator applied the new label to the ManagedCluster
```
Comment (bartwensley) [inline docs/cluster_configuration.md:205]: nit: applies

File: docs/cluster_configuration.md [hunk 8]
```diff
@@ -177,3 +177,83 @@ status:
...
+5. The ACM Policy Propagator detects the new binding:
+    * The old policies created through the [sno-ran-du-pg-v4-16-v1](#./git-setup/policytemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v1.yaml) Policy Generator are no longer matched to the ManagedCluster.
+    * The new policies created through the [sno-ran-du-pg-v4-16-v2](#./git-setup/policytemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v2.yaml) Policy Generator are matched to the ManagedCluster.
+    * The `ConfigurationApplied` condition is updated in the ProvisioningRequest to show that the configuration has changes and is being applied (the policies are dependend upon each other, so some are in a `Pending` state until ACM confirms their compliance):
```
Comment (bartwensley) [inline docs/cluster_configuration.md:209]: nit: changed
nit: "the policies depend on each other"

File: docs/samples/git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v2.yaml [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+        type: object
+
+# Notes:
+# clusterInstanceSchema contains only params that are exposed to ClusterRequest.
+#
+# When HW is ready, bmcAddress, bmcCredentialsName, bootMACAddress and nodes.nodeNetwork.macAddress
+# should be removed from the schema as they are supposed to come from HW.
+#
+# Ideally, rootDeviceHints should come from default configmap. Keep it in the schema to make this template
+# be able to used for mulitple SNOs with different rootDeviceHints.
```
Comment (bartwensley) [inline docs/samples/git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v2.yaml:289]: nit: multiple

**Review by bartwensley — CHANGES_REQUESTED** [review]
Looks good Irina - thanks

File: docs/cluster_configuration.md [hunk 10]
```diff
@@ -177,3 +177,83 @@ status:
     policyNamespace: ztp-sno-ran-du-v4-16
     remediationAction: enforce
 ```
+
+### Updates to an existing ACM PolicyGenerator manifest
+
+For updating a manifest in an existing ACM PolicyGenerator, the following steps need to be taken (we'll take [sno-ran-du-pg-v4-16-v1](#./git-setup/policytemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v1.yaml) for examples):
```
[the remark this answers is not in the document]
Comment (irinamihai) [reply docs/cluster_configuration.md]: I thought there was an issue with my vscode. They're all fine now, apparently the `#` wasn't needed either. Thanks for checking!

File: docs/cluster_configuration.md (hunk 5 above)
Comment (irinamihai) [reply docs/cluster_configuration.md]: I updated the names for the YAML files holding the ClusterTemplates.

File: docs/samples/git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v4-Y-Z-2.yaml [hunk 11]
```diff
@@ -0,0 +1,289 @@
...
+            type: string
+          hugepages-default:
+            type: string
+          hugepages-size:
+            type: string
+          hugepages-count:
+            type: string
+          install-plan-approval:
+            type: string
+            default: Automatic
```
Comment (Missxiaoguo) [inline docs/samples/git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v4-Y-Z-2.yaml:42]:
```suggestion
    type: object
```

File: docs/samples/git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v4-Y-Z-2.yaml [hunk 12]
```diff
@@ -0,0 +1,289 @@
...
+              provide access to instances. This key will be added to the host
+              to allow ssh access
+            type: string
+        required:
+        - clusterName
+        - nodes
+        type: object
+
+# Notes:
+# clusterInstanceSchema contains only params that are exposed to ClusterRequest.
```
Comment (Missxiaoguo) [inline docs/samples/git-setup/clustertemplates/version_4.Y.Z/sno-ran-du/sno-ran-du-v4-Y-Z-2.yaml:283]:
```suggestion
# clusterInstanceParameters contains only params that are exposed to ProvisioningRequest.
```

**Pull request merged**